src/manage_sql_secinfo.c

```diff
@@ -2656,6 +2657,485 @@ insert_cve_from_entry (element_t entry, element_t last_modified,
+  char * version_start_excl = NULL;
+  char * version_end_incl = NULL;
+  char * version_end_excl = NULL;
+
+  cJSON_ArrayForEach(match_rule, match_rules)
+    {
+      char *quoted_cpe;
+      vulnerable = FALSE;
+      cpe = NULL;
+      version_start_incl = NULL;
+      version_start_excl = NULL;
+      version_end_incl = NULL;
+      version_end_excl = NULL;
+
+      if (cJSON_IsTrue(cJSON_GetObjectItemCaseSensitive(match_rule, "vulnerable")))
+        vulnerable = TRUE;
+      else
+        vulnerable = FALSE;
+      cpe_js = cJSON_GetObjectItemCaseSensitive(match_rule, "cpe23Uri");
+      if (cpe_js != NULL)
+        cpe = cpe_js->valuestring;
+      quoted_cpe = sql_quote (cpe);
+      ver_se = cJSON_GetObjectItemCaseSensitive(match_rule, "versionStartIncluding");
+      if (ver_se != NULL)
+        version_start_incl = ver_se->valuestring;
+      ver_se = cJSON_GetObjectItemCaseSensitive(match_rule, "versionStartExcluding");
+      if (ver_se != NULL)
+        version_start_excl = ver_se->valuestring;
+      ver_se = cJSON_GetObjectItemCaseSensitive(match_rule, "versionEndIncluding");
+      if (ver_se != NULL)
+        version_end_incl = ver_se->valuestring;
+      ver_se = cJSON_GetObjectItemCaseSensitive(match_rule, "versionEndExcluding");
+      if (ver_se != NULL)
+        version_end_excl = ver_se->valuestring;
+
+      sql
+       ("INSERT INTO scap2.cpe_match_range"
+        " (node_id, vulnerable, cpe,"
+	"  version_start_incl, version_start_excl,"
+	"  version_end_incl, version_end_excl)"
+        " VALUES"
+        " (%ld, %d, '%s', '%s', '%s', '%s', '%s')",
+        id,
+        vulnerable ? 1 : 0,
+	quoted_cpe,
+        version_start_incl,
+        version_start_excl,
+        version_end_incl,
+        version_end_excl);
+      g_free (quoted_cpe);
+    }
+}
+
+/**
+ * @brief Load and add recursively all nodes of a match rule tree for a
+ *        specific CVE. Build a match rule tree.
+ *
+ * @param[in]  parent_id  The parent_id of the nodes to insert
+ *                        (0 for the root node).
+ * @param[in]  cveid      The id of the CVE the tree belongs to.
+ * @param[in]  nodes      The JSON object that contains the rules for a
+ *                        specific tree level.
+ */
+static void
+load_nodes (long int parent_id, long int cveid, cJSON *nodes)
+{
+  cJSON *node;
+  long int id;
+  cJSON *operator;
+  cJSON *cpe_match_rules;
+  cJSON *child_nodes;
+
+  node = NULL;
+  id = 0;
+  operator = NULL;
+  cpe_match_rules = NULL;
+  child_nodes = NULL;
+
+  cJSON_ArrayForEach(node, nodes)
+    {
+      operator = cJSON_GetObjectItemCaseSensitive(node, "operator");
+      id = save_node (parent_id, cveid, operator->valuestring);
+      cpe_match_rules = cJSON_GetObjectItemCaseSensitive(node, "cpe_match");
+      add_cpe_match_rules (id, cpe_match_rules);
+      child_nodes = cJSON_GetObjectItemCaseSensitive(node, "children");
+      load_nodes (id, cveid, child_nodes);
+    }
+}
```

Comment:
I know the macro handles that but I would suggest checking `nodes` for NULL and returning if so.

Does the return of `cJSON_GetObjectItemCaseSensitive` need to be checked for NULL ?